### PR TITLE
Allow gprc endpoints to return streaming out.

### LINKED ISSUE #44: Return `stream` instead of `sequence` in the grpc example
The grpc example: https://github.com/arr-ai/arrai/tree/master/examples/grpc

## The current generated proto text
```protobuf
//   THIS IS AUTOGENERATED BY sysl   //
syntax = "proto3";

package github.com/foo/foo;

option go_package = "github.com/foo/foo";

message Customer {
    string customerID = 1;
    string dob = 2;
}

message Account {
    string accountID = 1;
    string opened = 2;
    double balance = 3;
}

service Foo {
    rpc Getcustomer() returns (sequence of Customer) {}
}

```

## The expected proto
There are two possible results depend on the user's intention:
1. 
```diff
+ message Customers {
+     repeated Customer customers = 1;
+ }

service Foo {
-     rpc Getcustomer() returns (sequence of Customer) {}
+     rpc Getcustomers() returns (Customers) {}
}
```
2. 
```diff
service Foo {
-     rpc Getcustomer() returns (sequence of Customer) {}
+     rpc Getcustomers() returns (stream Customer) {}
}
```

### PR DESCRIPTION
**Pull request by springwiz**
Fixes #44.

Changes proposed in this pull request:
- Tweak the grpc example to handle streaming out.

### CODE REVIEW

File: examples/grpc/proto-util.arrai [hunk 1]
```diff
@@ -10,11 +10,20 @@ let grpc = //./grpc;
     endpointInfo: \ep
         let method = //.str.sub(//.str.title(//.str.lower(ep.name)), "-", "");
         let paramName = method + "Request" if wrap else $"${ep.params >> grpc.type(.)::, }";
-        let responseName = method + "Response" if wrap else ep.ret("ok");
+        let streamRes = //.str.sub(ep.ret("ok"), "sequence of", "stream") if ep.attrs.stream else //.str.sub(ep.ret("ok"), "sequence of ", "") + "s";
```
Comment (marcelocantos) [inline examples/grpc/proto-util.arrai:13]: Would this suffice?
```suggestion
        let streamRes = //.str.sub(ep.ret("ok"), "sequence of ", "stream" if ep.attrs.stream) + "s";
```

File: examples/grpc/proto-util.arrai [hunk 2]
```diff
@@ -10,11 +10,20 @@ let grpc = //./grpc;
...
             responseName: responseName,
             rpcMethod:    $`rpc ${method}(${paramName}) returns (${responseName}) {}`,
         ),
+
+    wrapSequence: \ep
+        let wrapType = //.str.sub(ep.ret("ok"), "sequence of ", "") + "s";
+        let name = //.str.lower(//.str.sub(ep.ret("ok"), "sequence of ", "")) + "s";
+        let type = //.str.sub(ep.ret("ok"), "sequence of ", "");
+        (
+            grpcType: $`message ${wrapType} { repeated ${type} ${name} = 1; }` if !ep.attrs.stream else $``,
```
Comment (marcelocantos) [inline examples/grpc/proto-util.arrai:27]:
```
 else $``
```
not required

File: examples/grpc/grpc-proto.arrai [hunk 3]
```diff
@@ -34,6 +34,7 @@ let endpoints = app.endpoints where !//.str.has_prefix(.@item.name, "enum ");
                 enum ${//.rex.sub(.name, `^\S*\s`, "")} {
                     ${.stmts >> $`${.action};`::\i}
                 }`:::\n}
+            ${endpoints >> (proto.wrapSequence(.)).grpcType::\i}
```
Comment (marcelocantos) [inline examples/grpc/grpc-proto.arrai:37]: This should work now:
```suggestion
            ${endpoints >> proto.wrapSequence(.).grpcType::\i}
```

File: examples/grpc/proto-util.arrai [hunk 4]
```diff
@@ -10,11 +10,20 @@ let grpc = //./grpc;
...
+        let responseName = method + "Response" if wrap else streamRes;
         (
             method:       method,
             paramName:    paramName,
             responseName: responseName,
             rpcMethod:    $`rpc ${method}(${paramName}) returns (${responseName}) {}`,
         ),
+
+    wrapSequence: \ep
+        let wrapType = //.str.sub(ep.ret("ok"), "sequence of ", "") + "s";
```
Comment (marcelocantos) [inline examples/grpc/proto-util.arrai:23]: `//.str.sub(ep.ret("ok"), "sequence of ", "")` occurs a fair bit. Should refactor.

File: examples/grpc/proto-util.arrai (hunk 1 above)
Comment (springwiz) [reply examples/grpc/proto-util.arrai:13]: Not really this appends "s" for both if and else, but I want it only for else.

File: examples/grpc/proto-util.arrai (hunk 2 above)
Comment (springwiz) [reply examples/grpc/proto-util.arrai:27]: ok

File: examples/grpc/grpc-proto.arrai (hunk 3 above)
Comment (springwiz) [reply examples/grpc/grpc-proto.arrai:37]: ok, works with the latest version of arrai.

File: examples/grpc/proto-util.arrai (hunk 4 above)
Comment (springwiz) [reply examples/grpc/proto-util.arrai:23]: ok

**Pull request merged**